### PR TITLE
[Doc] Fix the linkcheck CI job

### LINKED ISSUE #41136: [docs infra] Linkcheck failures are noisy and hurt dev productivity
### Description

Our linkcheck has been broken for some time. Developers need to scroll through false errors to find the ones they may have caused. Some developers want to turn off linkcheck all together, which would be bad for user experience. We should be able to make linkcheck more robust. 
cc:  @peytondmurray 

@can-anyscale agrees to look into this for the sprint beginning Nov 15.

Here are some examples.

```
(ray-core/examples/map_reduce: line 180002) broken    https://medium.com/distributed-computing-with-ray/executing-adistributed-shuffle-without-a-mapreduce-system-d5856379426c - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/executing-a-distributed-shuffle-without-a-mapreduce-system-d5856379426c
--
  | 2023-11-09 17:35:47 UTC | (ray-overview/examples: line  587) broken    https://medium.com/distributed-computing-with-ray/intro-to-rllib-example-environments-3a113f532c70 - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/intro-to-rllib-example-environments-3a113f532c70
  | 2023-11-09 17:35:47 UTC | (     serve/index: line  295) broken    https://medium.com/distributed-computing-with-ray/how-to-scale-up-your-fastapi-application-using-ray-serve-c9a7b69e786 - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/how-to-scale-up-your-fastapi-application-using-ray-serve-c9a7b69e786
  | 2023-11-09 17:35:47 UTC | (ray-overview/getting-started: line   23) broken    https://medium.com/distributed-computing-with-ray - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray
  | 2023-11-09 17:35:47 UTC | (cluster/vms/user-guides/launching-clusters/aws: line   65) broken    https://medium.com/distributed-computing-with-ray/a-step-by-step-guide-to-scaling-your-first-python-application-in-the-cloud-8761fe331ef1 - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/a-step-by-step-guide-to-scaling-your-first-python-application-in-the-cloud-8761fe331ef1
  | 2023-11-09 17:35:47 UTC | (ray-overview/examples: line  617) broken    https://medium.com/distributed-computing-with-ray/lessons-from-implementing-12-deep-rl-algorithms-in-tf-and-pytorch-1b412009297d - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/lessons-from-implementing-12-deep-rl-algorithms-in-tf-and-pytorch-1b412009297d
  | 2023-11-09 17:35:47 UTC | (ray-overview/examples: line  611) broken    https://medium.com/distributed-computing-with-ray/reinforcement-learning-with-rllib-in-the-unity-game-engine-1a98080a7c0d - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/reinforcement-learning-with-rllib-in-the-unity-game-engine-1a98080a7c0d
  | 2023-11-09 17:35:47 UTC | (     serve/index: line  293) broken    https://medium.com/distributed-computing-with-ray/the-simplest-way-to-serve-your-nlp-model-in-production-with-pure-python-d42b6a97ad55 - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/the-simplest-way-to-serve-your-nlp-model-in-production-with-pure-python-d42b6a97ad55
  | 2023-11-09 17:35:47 UTC | (     serve/index: line  294) broken    https://medium.com/distributed-computing-with-ray/machine-learning-serving-is-broken-f59aff2d607f - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/machine-learning-serving-is-broken-f59aff2d607f
  | 2023-11-09 17:35:47 UTC | (ray-overview/ray-libraries: line  159) broken    https://medium.com/ludwig-ai/ludwig-ai-v0-4-introducing-declarative-mlops-with-ray-dask-tabnet-and-mlflow-integrations-6509c3875c2e - 429 Client Error: Too Many Requests for url: https://medium.com/ludwig-ai/ludwig-ai-v0-4-introducing-declarative-mlops-with-ray-dask-tabnet-and-mlflow-integrations-6509c3875c2e
  | 2023-11-09 17:35:47 UTC | (ray-overview/examples: line  629) broken    https://medium.com/riselab/functional-rl-with-keras-and-tensorflow-eager-7973f81d6345 - 429 Client Error: Too Many Requests for url: https://medium.com/riselab/functional-rl-with-keras-and-tensorflow-eager-7973f81d6345
  | 2023-11-09 17:35:47 UTC | (ray-overview/getting-started: line   14) broken    https://medium.com/riselab/rayonspark-running-emerging-ai-applications-on-big-data-clusters-with-ray-and-analytics-zoo-923e0136ed6a - 429 Client Error: Too Many Requests for url: https://medium.com/riselab/rayonspark-running-emerging-ai-applications-on-big-data-clusters-with-ray-and-analytics-zoo-923e0136ed6a
  | 2023-11-09 17:35:47 UTC | (ray-overview/getting-started: line   17) broken    https://medium.com/riselab/cutting-edge-hyperparameter-tuning-with-ray-tune-be6c0447afdf - 429 Client Error: Too Many Requests for url: https://medium.com/riselab/cutting-edge-hyperparameter-tuning-with-ray-tune-be6c0447afdf
  | 2023-11-09 17:35:47 UTC | (ray-overview/examples: line  605) broken    https://medium.com/distributed-computing-with-ray/attention-nets-and-more-with-rllibs-trajectory-view-api-d326339a6e65 - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/attention-nets-and-more-with-rllibs-trajectory-view-api-d326339a6e65
  | 2023-11-09 17:35:59 UTC | -rate limited-   https://towardsdatascience.com/doing-xgboost-hyper-parameter-tuning-the-smart-way-part-1-of-2-f6d255a45dde \| sleeping...
  | 2023-11-09 17:35:59 UTC | -rate limited-   https://towardsdatascience.com/ensemble-methods-bagging-boosting-and-stacking-c9214a10a205 \| sleeping...
  | 2023-11-09 17:36:00 UTC | (tune/examples/tune-pytorch-lightning: line 10012) broken    https://towardsdatascience.com/from-pytorch-to-pytorch-lightning-a-gentle-introduction-b371b7caaf09 - 429 Client Error: Too Many Requests for url: https://towardsdatascience.com/from-pytorch-to-pytorch-lightning-a-gentle-introduction-b371b7caaf09
  | 2023-11-09 17:36:00 UTC | (ray-overview/getting-started: line   16) broken    https://towardsdatascience.com/fast-hyperparameter-tuning-at-scale-d428223b081c - 429 Client Error: Too Many Requests for url: https://towardsdatascience.com/fast-hyperparameter-tuning-at-scale-d428223b081c
  | 2023-11-09 17:36:00 UTC | (ray-overview/getting-started: line    8) broken    https://towardsdatascience.com/modern-parallel-and-distributed-python-a-quick-tutorial-on-ray-99f8d70369b8 - 429 Client Error: Too Many Requests for url: https://towardsdatascience.com/modern-parallel-and-distributed-python-a-quick-tutorial-on-ray-99f8d70369b8
  | 2023-11-09 17:41:00 UTC | (tune/examples/tune-xgboost: line 110023) broken    https://towardsdatascience.com/doing-xgboost-hyper-parameter-tuning-the-smart-way-part-1-of-2-f6d255a45dde - 429 Client Error: Too Many Requests for url: https://towardsdatascience.com/doing-xgboost-hyper-parameter-tuning-the-smart-way-part-1-of-2-f6d255a45dde
  | 2023-11-09 17:41:01 UTC | (tune/examples/tune-xgboost: line 10052) ok        https://towardsdatascience.com/ensemble-methods-bagging-boosting-and-stacking-c9214a10a205
  | 2023-11-09 17:41:01 UTC | build finished with problems, 11 warnings.
  | 2023-11-09 17:41:01 UTC | sphinx-sitemap warning: No pages generated for sitemap.xml
  | 2023-11-09 17:41:01 UTC |  
  | 2023-11-09 17:41:01 UTC | BROKEN LINKS SUMMARY:
  | 2023-11-09 17:41:01 UTC |  
  | 2023-11-09 17:41:01 UTC | (rllib/rllib-concepts: line  406) broken    https://github.com/ray-project/ray/blob/master/rllib/algorithms/a3c/a3c_torch_policy.py - 404 Client Error: Not Found for url: https://github.com/ray-project/ray/blob/master/rllib/algorithms/a3c/a3c_torch_policy.py
  | 2023-11-09 17:41:01 UTC | (tune/examples/tune_mnist_keras: line 10006) broken    https://keras.io - HTTPSConnectionPool(host='keras.io', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1131)')))
  | 2023-11-09 17:41:01 UTC | (rllib/rllib-torch2x: line    8) broken    https://pytorch.org/docs/stable/dynamo/index.html#torchdynamo-overview - 404 Client Error: Not Found for url: https://pytorch.org/docs/stable/dynamo/index.html
  | 2023-11-09 17:41:01 UTC | (ray-core/examples/map_reduce: line 180002) broken    https://medium.com/distributed-computing-with-ray/executing-adistributed-shuffle-without-a-mapreduce-system-d5856379426c - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/executing-a-distributed-shuffle-without-a-mapreduce-system-d5856379426c
  | 2023-11-09 17:41:01 UTC | (ray-overview/examples: line  587) broken    https://medium.com/distributed-computing-with-ray/intro-to-rllib-example-environments-3a113f532c70 - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/intro-to-rllib-example-environments-3a113f532c70
  | 2023-11-09 17:41:01 UTC | (     serve/index: line  295) broken    https://medium.com/distributed-computing-with-ray/how-to-scale-up-your-fastapi-application-using-ray-serve-c9a7b69e786 - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/how-to-scale-up-your-fastapi-application-using-ray-serve-c9a7b69e786
  | 2023-11-09 17:41:01 UTC | (ray-overview/getting-started: line   23) broken    https://medium.com/distributed-computing-with-ray - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray
  | 2023-11-09 17:41:01 UTC | (cluster/vms/user-guides/launching-clusters/aws: line   65) broken    https://medium.com/distributed-computing-with-ray/a-step-by-step-guide-to-scaling-your-first-python-application-in-the-cloud-8761fe331ef1 - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/a-step-by-step-guide-to-scaling-your-first-python-application-in-the-cloud-8761fe331ef1
  | 2023-11-09 17:41:01 UTC | (ray-overview/examples: line  617) broken    https://medium.com/distributed-computing-with-ray/lessons-from-implementing-12-deep-rl-algorithms-in-tf-and-pytorch-1b412009297d - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/lessons-from-implementing-12-deep-rl-algorithms-in-tf-and-pytorch-1b412009297d
  | 2023-11-09 17:41:01 UTC | (ray-overview/examples: line  611) broken    https://medium.com/distributed-computing-with-ray/reinforcement-learning-with-rllib-in-the-unity-game-engine-1a98080a7c0d - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/reinforcement-learning-with-rllib-in-the-unity-game-engine-1a98080a7c0d
  | 2023-11-09 17:41:01 UTC | (     serve/index: line  293) broken    https://medium.com/distributed-computing-with-ray/the-simplest-way-to-serve-your-nlp-model-in-production-with-pure-python-d42b6a97ad55 - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/the-simplest-way-to-serve-your-nlp-model-in-production-with-pure-python-d42b6a97ad55
  | 2023-11-09 17:41:01 UTC | (     serve/index: line  294) broken    https://medium.com/distributed-computing-with-ray/machine-learning-serving-is-broken-f59aff2d607f - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/machine-learning-serving-is-broken-f59aff2d607f
  | 2023-11-09 17:41:01 UTC | (ray-overview/ray-libraries: line  159) broken    https://medium.com/ludwig-ai/ludwig-ai-v0-4-introducing-declarative-mlops-with-ray-dask-tabnet-and-mlflow-integrations-6509c3875c2e - 429 Client Error: Too Many Requests for url: https://medium.com/ludwig-ai/ludwig-ai-v0-4-introducing-declarative-mlops-with-ray-dask-tabnet-and-mlflow-integrations-6509c3875c2e
  | 2023-11-09 17:41:01 UTC | (ray-overview/examples: line  629) broken    https://medium.com/riselab/functional-rl-with-keras-and-tensorflow-eager-7973f81d6345 - 429 Client Error: Too Many Requests for url: https://medium.com/riselab/functional-rl-with-keras-and-tensorflow-eager-7973f81d6345
  | 2023-11-09 17:41:01 UTC | (ray-overview/getting-started: line   14) broken    https://medium.com/riselab/rayonspark-running-emerging-ai-applications-on-big-data-clusters-with-ray-and-analytics-zoo-923e0136ed6a - 429 Client Error: Too Many Requests for url: https://medium.com/riselab/rayonspark-running-emerging-ai-applications-on-big-data-clusters-with-ray-and-analytics-zoo-923e0136ed6a
  | 2023-11-09 17:41:01 UTC | (ray-overview/getting-started: line   17) broken    https://medium.com/riselab/cutting-edge-hyperparameter-tuning-with-ray-tune-be6c0447afdf - 429 Client Error: Too Many Requests for url: https://medium.com/riselab/cutting-edge-hyperparameter-tuning-with-ray-tune-be6c0447afdf
  | 2023-11-09 17:41:01 UTC | (ray-overview/examples: line  605) broken    https://medium.com/distributed-computing-with-ray/attention-nets-and-more-with-rllibs-trajectory-view-api-d326339a6e65 - 429 Client Error: Too Many Requests for url: https://medium.com/distributed-computing-with-ray/attention-nets-and-more-with-rllibs-trajectory-view-api-d326339a6e65
  | 2023-11-09 17:41:01 UTC | (tune/examples/tune-pytorch-lightning: line 10012) broken    https://towardsdatascience.com/from-pytorch-to-pytorch-lightning-a-gentle-introduction-b371b7caaf09 - 429 Client Error: Too Many Requests for url: https://towardsdatascience.com/from-pytorch-to-pytorch-lightning-a-gentle-introduction-b371b7caaf09
  | 2023-11-09 17:41:01 UTC | (ray-overview/getting-started: line   16) broken    https://towardsdatascience.com/fast-hyperparameter-tuning-at-scale-d428223b081c - 429 Client Error: Too Many Requests for url: https://towardsdatascience.com/fast-hyperparameter-tuning-at-scale-d428223b081c
  | 2023-11-09 17:41:01 UTC | (ray-overview/getting-started: line    8) broken    https://towardsdatascience.com/modern-parallel-and-distributed-python-a-quick-tutorial-on-ray-99f8d70369b8 - 429 Client Error: Too Many Requests for url: https://towardsdatascience.com/modern-parallel-and-distributed-python-a-quick-tutorial-on-ray-99f8d70369b8
  | 2023-11-09 17:41:01 UTC | (tune/examples/tune-xgboost: line 110023) broken    https://towardsdatascience.com/doing-xgboost-hyper-parameter-tuning-the-smart-way-part-1-of-2-f6d255a45dde - 429 Client Error: Too Many Requests for url: https://towardsdatascience.com/doing-xgboost-hyper-parameter-tuning-the-smart-way-part-1-of-2-f6d255a45dde

```

### Link

[Sample linkcheck log](https://buildkite.com/ray-project/oss-ci-build-pr/builds/41227#018bb513-f885-472d-8858-63669d2ce871)

### PR DESCRIPTION
**Pull request by peytondmurray**
## Why are these changes needed?

- This PR fixes the linkcheck job by bypassing `ci.sh`, and using a docker container on the buildkite runner.
- This PR also fixes a ton of broken links that were causing linkcheck to fail (even without rate limit issues), mostly by removing links to examples that were deleted.
- I also removed the global `depends_on` in `lint.rayci.yml`; not all lint jobs require the same environment.

## Related issue number

Closes #41136.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .buildkite/pipeline.test.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
...
-  instance_size: small
-  commands:
-    - export LINT=1
-    - ./ci/env/install-dependencies.sh
-    - ./ci/ci.sh check_sphinx_links
-  soft_fail: True
+name: ":book: LinkCheck"
+description: "Check the integrity of all external doc links with sphinx"
+steps:
+  - label: "Run linkcheck"
```
Comment (can-anyscale) [inline .buildkite/pipeline.test.yml:4]: can we move this to lint.rayci.yaml or build.rayci.yml, the pipeline.test.yml is on CIv1 and doesn't have the job_env 

also I don't think the `description` field is supported

our CI only supports a selected set of buildkite field due to security reason

File: doc/source/ray-overview/examples.rst [hunk 2]
```diff
@@ -619,22 +619,10 @@ Ray Examples
 
     .. grid-item-card:: :bdg-secondary:`Code example`
         :class-item: gallery-item reinforcement-learning rllib
-        :link: https://github.com/ray-project/ray/blob/master/rllib/examples/batch_norm_model.py
+        :link: https://github.com/ray-project/ray/blob/master/rllib/examples/models/batch_norm_model.py
```
Comment (aslonnie) [inline doc/source/ray-overview/examples.rst:622]: these are two different files. the original one (already deleted) has a `__main__` and is directly runnable, where this one is not.

same with many other ones.

Comment (aslonnie) [reply doc/source/ray-overview/examples.rst:616]: I am not really the owner

@sven1977 ?

File: doc/source/ray-contribute/docs.ipynb [hunk 3]
```diff
@@ -232,9 +231,10 @@
     "Sometimes you might want to add a completely new document to the Ray documentation, like adding a new\n",
     "user guide or a new example.\n",
     "\n",
-    "For this to work, you need to make sure to add the new document explicitly to the \n",
-    "[`_toc.yml` file](https://github.com/ray-project/ray/blob/master/doc/source/_toc.yml) that determines\n",
-    "the structure of the Ray documentation.\n",
+    "For this to work, you need to make sure to add the new document explicitly to a parent document's toctree \n",
```
Comment (angelinalg) [inline doc/source/ray-contribute/docs.ipynb:234]:
```suggestion
    "For this to work, you need to make sure to add the new document explicitly to a parent document's toctree, \n",
```

File: doc/source/ray-contribute/docs.ipynb [hunk 4]
```diff
@@ -232,9 +231,10 @@
...
+    "which will determine the structure of the Ray documentation. See \n",
```
Comment (angelinalg) [inline doc/source/ray-contribute/docs.ipynb:235]:
```suggestion
    "which determines the structure of the Ray documentation. See \n",
```

**Review by architkulkarni — APPROVED** [review]
Thanks for doing this! `kubernetes` change looks good

File: doc/source/ray-overview/ray-libraries.rst [hunk 5]
```diff
@@ -11,19 +11,8 @@ more information.
...
-
-          - name: the integration link button text
-            section_title: The section title for this integration
-            description: A quick description of your library and its integration with Ray
-            website: The URL of your website
-            repo: The URL of your project on GitHub
-            image: The URL of a logo of your project
-
-    That's all!
+    To add an integration simply add an entry to this file, using the same
```
Comment (angelinalg) [inline doc/source/ray-overview/ray-libraries.rst:14]:
```suggestion
    To add an integration add an entry to this file, using the same
```

File: doc/source/ray-overview/ray-libraries.rst [hunk 6]
```diff
@@ -11,19 +11,8 @@ more information.
...
+    ``grid-item-card`` directive that the other examples use. That's all!
```
Comment (angelinalg) [inline doc/source/ray-overview/ray-libraries.rst:15]:
```suggestion
    ``grid-item-card`` directive that the other examples use.
```

File: doc/source/rllib/rllib-concepts.rst [hunk 7]
```diff
@@ -164,7 +164,7 @@ We can create an `Algorithm <#algorithms>`__ and try running this policy on a to
     tune.Tuner(MyAlgo, param_space={"env": "CartPole-v1", "num_workers": 2}).fit()
 
 
-If you run the above snippet `(runnable file here) <https://github.com/ray-project/ray/blob/master/rllib/examples/custom_tf_policy.py>`__, you'll probably notice that CartPole doesn't learn so well:
+If you run the above snippet, you'll probably notice that CartPole doesn't learn so well:
```
Comment (angelinalg) [inline doc/source/rllib/rllib-concepts.rst:167]:
```suggestion
If you run the above snippet, notice that CartPole doesn't learn so well:
```

File: doc/source/rllib/rllib-dev.rst [hunk 8]
```diff
@@ -111,7 +111,7 @@ Keeping the memory usage of long running workers stable can be challenging. The
 
 .. autoclass:: ray.rllib.algorithms.callbacks.MemoryTrackingCallbacks
 
-The objects with the top 20 memory usage in the workers will be added as custom metrics. These can then be monitored using tensorboard or other metrics integrations like Weights and Biases: 
+The objects with the top 20 memory usage in the workers will be added as custom metrics. These can then be monitored using tensorboard or other metrics integrations like Weights and Biases:
```
Comment (angelinalg) [inline doc/source/rllib/rllib-dev.rst:114]:
```suggestion
The objects with the top 20 memory usage in the workers are added as custom metrics. These can then be monitored using tensorboard or other metrics integrations like Weights & Biases:
```

**Review by angelinalg — APPROVED** [review]

File: .buildkite/build.rayci.yml [hunk 9]
```diff
@@ -160,3 +160,9 @@ steps:
     soft_fail: true
     depends_on:
       - forge
+
+  - label: ":book: LinkCheck"
```
Comment (can-anyscale) [inline .buildkite/build.rayci.yml:164]: nit: now see this again, can me move it to lint.rayci.yml; not really a build; also maybe follow the existing name format from the same file, thankks

**Review by can-anyscale — APPROVED** [review]
ci change looks good!

File: .buildkite/build.rayci.yml (hunk 9 above)
Comment (peytondmurray) [reply .buildkite/build.rayci.yml:164]: Will do now :+1:

**Review by ArturNiederfahrenhorst — APPROVED** [review]
lgtm. I checked the RLlib related changes.

**Comment by peytondmurray** [comment]
RTD looks like it failed here, but if you click on the link it looks like the docs built successfully :/. It also clearly didn't reach the 1h time limit, not sure what's happening here. cc @angelinalg I know you've seen this before as well.

**Pull request merged**